checkers/rules/rules.go

```diff
@@ -736,3 +736,34 @@ func dynamicFmtString(m dsl.Matcher) {
 		Suggest("errors.New($f($*args))").
 		Report(`use errors.New($f($*args)) or fmt.Errorf("%s", $f($*args)) instead`)
 }
+
+//doc:summary Detects strings.Compare usage
+//doc:tags    performance experimental
+//doc:before  strings.Compare(x, y)
+//doc:after   x < y
+func stringsCompare(m dsl.Matcher) {
+	m.Match(`strings.Compare($s1, $s2) == 0`).
+		Report(`don't use strings.Compare on hot path, change it to built-in operators`).
```

Comment:
This warning message is not helpful.
Suggest template is more explanatory.
In general, the warning message should tell how to fix the situation.
In other words, it should tell what to do instead of saying what not to do.

Reply:
Maybe this message comes from go-perfguard style warnings.
But there is a difference there: it doesn't report anything if it has a replacement, so it just reports the suggestion template there. If there is no suggestion available, it can only say "don't do that" because it doesn't know how to fix the code.

In my opinion, reporting performance issue without a fix is only useful if it's a big performance issue.

Reply:
\+